added functions to disable/enable hot reload

NikAiyer opened the pull request:
## Description

<!-- Provide a brief description of the changes in this PR -->
This PR adds function to enable the disabling of hot reloading for certain server handlers.


## Related Issue(s)

<!-- Link to the issue(s) this PR addresses, using hashtag notation: #123 -->

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test update

## Checklist

- [ ] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works

NikAiyer committed:
added functions to disable/enable hot reload

DanielSLew commented on packages/deployer/src/server/handlers/client.ts:
```diff
@@ -34,3 +35,18 @@ export function handleTriggerClientsRefresh(c: Context) {
+
+// Functions to control hot reload during template installation
+export function disableHotReload() {
+  isTemplateInstallationActive = true;
```
I think we should make this more generalized. 
```suggestion
  isHotReadloadDisabled = true;
```

DanielSLew committed:
fix(core): pipe providerOptions in streamVNext, fix reasoning-delta (#7323)

LekoArts committed:
chore(docs): Add note about Nx monorepo limitation (#7255)

## Description

This PR clarifies our documentation about Nx + monorepos with Mastra

## Related Issue(s)

<!-- Link to the issue(s) this PR addresses, using hashtag notation:
#123 -->

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to change)
- [x] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test update

## Checklist

- [x] I have made corresponding changes to the documentation (if
applicable)
- [ ] I have added tests that prove my fix is effective or that my
feature works

epinzur committed:
Piped tracingContext through everything (#7274)

TylerBarnes committed:
ci: remove verdaccio auth to make e2e tests less flaky (#7322)

~~The e2e tests were failing intermittently with npm authentication
errors when publishing packages to the local verdaccio registry. Looking
at the CI history, the same commit would sometimes pass and sometimes
fail, indicating a race condition.~~

~~Added retry logic with exponential backoff to the publish command - if
publishing fails (typically due to auth issues), it'll retry up to 5
times with increasing wait times between attempts. This should handle
the race condition where the auth setup might not be fully ready when we
try to publish.~~

Retries didn't help, I ended up removing verdaccio auth entirely since
it doesn't matter for e2e tests

TylerBarnes committed:
fix(@mastra/core): prevent filtering out initial user message (#7317)

Might fix https://github.com/mastra-ai/mastra/issues/7231

Likely the recent change that removed the `.` from the default message
caused the empty user content message to be filtered out

wardpeet committed:
Add new @mastra/ai-sdk pkg (#7263)

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>
Co-authored-by: Lennart <lekoarts@gmail.com>

DanielSLew committed:
fix(pg): dont recreate existing index (#7326)

## Description

Fixes https://github.com/mastra-ai/mastra/issues/7264
<!-- Provide a brief description of the changes in this PR -->

## Related Issue(s)

<!-- Link to the issue(s) this PR addresses, using hashtag notation:
#123 -->

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to change)
- [ ] Documentation update
- [ ] Code refactoring
- [ ] Performance improvement
- [ ] Test update

## Checklist

- [ ] I have made corresponding changes to the documentation (if
applicable)
- [ ] I have added tests that prove my fix is effective or that my
feature works

TylerBarnes committed:
fix(@mastra/core): include assistant messages in onFinish callback for aisdk format (#7327)

Fixes #7254 where the onFinish callback wasn't returning assistant
messages when using format: 'aisdk' in streamVNext.

The issue was that while the messageList was being updated with response
messages, these weren't being passed to the user's onFinish callback.

Before the fix, when using streamVNext with format: 'aisdk', the
onFinish callback would wouldn't receive all response messages

```typescript
await agent.streamVNext('Hello', {
  format: 'aisdk',
  onFinish: (props) => {
    console.log(props.messages); // Missing assistant messages
  }
});
```

After the fix, the onFinish callback now receives all response messages

alliehowe29 committed:
fix(mcp-docs-server): Validate file path after join (#7238)

Co-authored-by: Allie Howe <ahowe@Allies-MacBook-Pro.local>
Co-authored-by: LekoArts <lekoarts@gmail.com>
Co-authored-by: Abhi Aiyer <abhiaiyer91@gmail.com>

wardpeet committed:
E2E fixes (#7329)

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

dane-ai-mastra[bot] committed:
chore(deps): update dependency rollup to ~4.49.0 (#7000)

Co-authored-by: dane-ai-mastra[bot] <199668360+dane-ai-mastra[bot]@users.noreply.github.com>

dane-ai-mastra[bot] committed:
chore(deps): update assistant ui (#5817)

Co-authored-by: dane-ai-mastra[bot] <199668360+dane-ai-mastra[bot]@users.noreply.github.com>
Co-authored-by: Ward Peeters <ward@coding-tech.com>

dane-ai-mastra[bot] committed:
chore(deps): update e2e tests (#7084)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
|
[@types/react](https://redirect.github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react)
([source](https://redirect.github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react))
| devDependencies | patch | [`^19.1.11` ->
`^19.1.12`](https://renovatebot.com/diffs/npm/@types%2freact/19.1.11/19.1.12)
|
|
[@types/react-dom](https://redirect.github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-dom)
([source](https://redirect.github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react-dom))
| devDependencies | patch | [`^19.1.8` ->
`^19.1.9`](https://renovatebot.com/diffs/npm/@types%2freact-dom/19.1.8/19.1.9)
|
| [hono](https://hono.dev)
([source](https://redirect.github.com/honojs/hono)) | dependencies |
patch | [`^4.9.4` ->
`^4.9.5`](https://renovatebot.com/diffs/npm/hono/4.9.4/4.9.5) |

---

### Release Notes

<details>
<summary>honojs/hono (hono)</summary>

###
[`v4.9.5`](https://redirect.github.com/honojs/hono/releases/tag/v4.9.5)

[Compare
Source](https://redirect.github.com/honojs/hono/compare/v4.9.4...v4.9.5)

#### What's Changed

- chore: replace supertest with undici by
[@&#8203;BarryThePenguin](https://redirect.github.com/BarryThePenguin)
in [#&#8203;4365](https://redirect.github.com/honojs/hono/pull/4365)
- fix(aws-lambda): preserve percent-encoded values in query strings by
[@&#8203;yusukebe](https://redirect.github.com/yusukebe) in
[#&#8203;4372](https://redirect.github.com/honojs/hono/pull/4372)
- feat(cors): Allow async functions for `origin` and `allowMethods` by
[@&#8203;jobrk](https://redirect.github.com/jobrk) in
[#&#8203;4373](https://redirect.github.com/honojs/hono/pull/4373)
- feat(cors): Correct origin function return type asynchronously
returning null or undefined for origin by
[@&#8203;jobrk](https://redirect.github.com/jobrk) in
[#&#8203;4375](https://redirect.github.com/honojs/hono/pull/4375)
- fix(service-worker): correct args for `app.fetch` in `handle` by
[@&#8203;yusukebe](https://redirect.github.com/yusukebe) in
[#&#8203;4374](https://redirect.github.com/honojs/hono/pull/4374)
- fix(language-detector): Detect language from path after getPath
changed by [@&#8203;iflamed](https://redirect.github.com/iflamed) in
[#&#8203;4369](https://redirect.github.com/honojs/hono/pull/4369)

#### New Contributors

- [@&#8203;jobrk](https://redirect.github.com/jobrk) made their first
contribution in
[#&#8203;4373](https://redirect.github.com/honojs/hono/pull/4373)
- [@&#8203;iflamed](https://redirect.github.com/iflamed) made their
first contribution in
[#&#8203;4369](https://redirect.github.com/honojs/hono/pull/4369)

**Full Changelog**:
<https://github.com/honojs/hono/compare/v4.9.4...v4.9.5>

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Enabled.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

👻 **Immortal**: This PR will be recreated if closed unmerged. Get
[config
help](https://redirect.github.com/renovatebot/renovate/discussions) if
that's undesired.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://redirect.github.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiI0MS44Ny40IiwidXBkYXRlZEluVmVyIjoiNDEuOTEuMyIsInRhcmdldEJyYW5jaCI6Im1haW4iLCJsYWJlbHMiOltdfQ==-->

---------

Co-authored-by: dane-ai-mastra[bot] <199668360+dane-ai-mastra[bot]@users.noreply.github.com>

NikAiyer committed:
Merge branch 'main' of https://github.com/mastra-ai/mastra into disable-hot-reload
